cmd/abapEnvironmentCheckoutBranch.go

```diff
@@ -74,38 +75,44 @@ func runAbapEnvironmentCheckoutBranch(options *abapEnvironmentCheckoutBranchOpti
 	client.SetOptions(clientOptions)
 	pollIntervall := com.GetPollIntervall()
 
-	log.Entry().Infof("Starting to switch branch to branch '%v' on repository '%v'", options.BranchName, options.RepositoryName)
-	log.Entry().Info("--------------------------------")
-	log.Entry().Info("Start checkout branch: " + options.BranchName)
-	log.Entry().Info("--------------------------------")
+	repositories := []abaputils.Repository{}
+	err = checkCheckoutBranchRepositoryConfiguration(*options)
 
-	// Triggering the Checkout of the repository into the ABAP Environment system
-	uriConnectionDetails, errorTriggerCheckout := triggerCheckout(options.RepositoryName, options.BranchName, connectionDetails, client)
-	if errorTriggerCheckout != nil {
-		log.Entry().WithError(errorTriggerCheckout).Fatal("Checkout of " + options.BranchName + " for software component " + options.RepositoryName + " failed on the ABAP System")
+	if err == nil {
+		repositories, err = abaputils.GetRepositories(&abaputils.RepositoriesConfig{BranchName: options.BranchName, RepositoryName: options.RepositoryName, Repositories: options.Repositories})
 	}
-
-	// Polling the status of the repository import on the ABAP Environment system
-	status, errorPollEntity := abaputils.PollEntity(options.RepositoryName, uriConnectionDetails, client, pollIntervall)
-	if errorPollEntity != nil {
-		log.Entry().WithError(errorPollEntity).Fatal("Status of checkout action on repository" + options.RepositoryName + " failed on the ABAP System")
+	if err == nil {
+		err = checkoutBranches(repositories, connectionDetails, client, pollIntervall)
 	}
-	if status == "E" {
-		log.Entry().Fatal("Checkout of branch " + options.BranchName + " failed on the ABAP System")
+	if err != nil {
+		return fmt.Errorf("Something failed during the checkout: %w", err)
```

Comment:
Same as for clone. Wrapping the error with "Something failed during the checkout:" does not provide valuable info (and would be misleading, if the error occurred in abaputils.GetRepositories)